lib/index.ts

```diff
@@ -28,12 +42,15 @@ const useBatchedState = <S = undefined>(initValue: S | (() => S)) => {
   const stateHookRes = useState<S>(initValue);
   const setter = stateHookRes[1];
 
-  function batchSetter(value: S | ((prevState: S) => S)) {
-    const setterUpdate: UpdateRecord = [setter, value];
+  const batchSetter = useCallback(
+    (value: S | ((prevState: S) => S)) => {
```

Comment:
There is `value: S | ((prevState: S) => S)`, which can be simplified to 
```ts
// Dispatch is imported from react
useCallback<Dispatch<SetStateAction<S>>>(value => { /* ... */ })
```
This way you can have the `type` of value without repeatedly typing `value: S | ((prevState: S) => S)`.

This doesn't change any js or ts, it just simplifies the types.